src/socketio/transport.rs

```diff
@@ -315,27 +314,32 @@ impl TransportClient {
                         // check which callback to use and call it with the data if it's present
                         if data.len() > 1 {
                             if let serde_json::Value::String(event) = contents[0].clone() {
-                                if let Some(function) = clone_self.get_event_callback(Event::Custom(event)) {
-                                    let lock = function.1.read().unwrap();
-                                    spawn_scoped!(lock(contents[1].to_string()));
-                                    drop(lock)
+                                if let Some(function) = clone_self.get_event_callback(&Event::Custom(event)) {
```

Comment:
Maybe refactor

Reply:
Maybe after some contributions, I'll have another look at this. I still don't love it, but its fine right now